src/Type/Accessory/AccessoryLowercaseStringType.php

```diff
@@ -326,6 +327,10 @@ public function isScalar(): TrinaryLogic
 
 	public function looseCompare(Type $type, PhpVersion $phpVersion): BooleanType
 	{
+		if ($type->isString()->yes() && $type->isLowercaseString()->no()) {
```

Comment:
Does https://3v4l.org/tEobn test pass (and vice versa)?

Reply:
Sounds like a very edge case which should be reported as a new issue.

I don't expect it works with this PR

Reply:
No, it's considered as false with the current implementation.

@staabm I think you'll need to add
```
&& ($type->isNumericString()->no() || $this->isNumericString()->no())
```
condition.

Reply:
It works already https://phpstan.org/r/3b02891b-463b-4d09-b845-428b10f44547 somehow ;-)

Reply:
I do not see any issue yet - https://phpstan.org/r/745f421e-ae6f-46f1-aa32-d2fe28d88b3f . But this PR will probably break it.

Reply:
You can cherry pick https://github.com/phpstan/phpstan-src/commit/e11991f2dffbf56f2f8c8be9652454547285e6b3 staabm

Reply:
picked and added a few more tests, thanks